chromium_src/chrome/browser/importer/importer_list.cc

```diff
@@ -47,30 +49,28 @@ void AddChromeToProfiles(std::vector<importer::SourceProfile>* profiles,
 void DetectChromeProfiles(std::vector<importer::SourceProfile>* profiles) {
   base::ScopedBlockingCall scoped_blocking_call(FROM_HERE,
                                                 base::BlockingType::WILL_BLOCK);
-  const base::FilePath chrome_user_data_folder = GetChromeUserDataFolder();
-  base::ListValue* chrome_profiles =
-      GetChromeSourceProfiles(chrome_user_data_folder);
-  const std::string brand_chrome("Chrome ");
-  AddChromeToProfiles(profiles, chrome_profiles, chrome_user_data_folder,
-                      brand_chrome);
-
+  AddChromeToProfiles(
+      profiles, GetChromeSourceProfiles(GetChromeUserDataFolder()),
+      GetChromeUserDataFolder(), "Chrome ", importer::TYPE_CHROME);
 #if !BUILDFLAG(IS_LINUX)
-  const base::FilePath canary_user_data_folder = GetCanaryUserDataFolder();
-  base::ListValue* canary_profiles =
-      GetChromeSourceProfiles(canary_user_data_folder);
-  const std::string brandCanary("Chrome Canary ");
-  AddChromeToProfiles(profiles, canary_profiles, canary_user_data_folder,
-                      brandCanary);
+  AddChromeToProfiles(
+      profiles, GetChromeSourceProfiles(GetCanaryUserDataFolder()),
+      GetCanaryUserDataFolder(), "Chrome Canary ", importer::TYPE_CHROME);
 #endif
+  AddChromeToProfiles(
+      profiles, GetChromeSourceProfiles(GetChromiumUserDataFolder()),
+      GetChromiumUserDataFolder(), "Chromium ", importer::TYPE_CHROME);
 
-  const base::FilePath chromium_user_data_folder = GetChromiumUserDataFolder();
-  base::ListValue* chromium_profiles =
-      GetChromeSourceProfiles(chromium_user_data_folder);
-  const std::string brandChromium("Chromium ");
-  AddChromeToProfiles(profiles, chromium_profiles, chromium_user_data_folder,
-                      brandChromium);
+  AddChromeToProfiles(
+      profiles, GetChromeSourceProfiles(GetEdgeUserDataFolder()),
+      GetEdgeUserDataFolder(), "Microsoft Edge ", importer::TYPE_EDGE);
```

Comment:
To avoid any conflicts with Legacy Edge (`importer::TYPE_EDGE`), I think we can make a new importer type (extend or patch `chrome/common/importer/importer_type.h`). Maybe a new value: `importer::TYPE_CHROMIUM_EDGE`.

We can then get rid of the patches here:
- `patches/chrome-browser-importer-importer_uma.cc.patch`
- `patches/chrome-common-importer-importer_type.h.patch`

Last would be updating `utility/importer/brave_profile_import_impl.cc` to use the new `importer::TYPE_CHROMIUM_EDGE`

Reply:
done